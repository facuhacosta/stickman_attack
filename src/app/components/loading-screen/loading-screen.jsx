import React, { useState } from 'react';
import style from './loading-screen.module.scss';
  
export function LoadingScreen({cameFrom, cameWith}) {

  const [visible, setVisible] = useState(true)
  return (
    <div className={`${style['loading-screen']}${visible ? '' : ` ${style['loading-screen-hidden']}`}`} onAnimationEnd={() => setVisible(false)} >
      {cameFrom === 'GAME' && cameWith && <p>{cameWith}</p>}
    </div>
  );
}