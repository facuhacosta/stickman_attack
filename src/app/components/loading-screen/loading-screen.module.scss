.loading-screen {
  background-color: grey;
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  animation: desapear 1s linear forwards;
  display: flex;
  justify-content: center;
  align-items: center;

  &-hidden {
    display: none;
  }

  & > p {
    font-size: 100px;
  }
}

@keyframes desapear {
  0% {
    opacity: 100;
  }

  50% {
    opacity: 100;
  }

  100% {
    opacity: 0;
  }
}