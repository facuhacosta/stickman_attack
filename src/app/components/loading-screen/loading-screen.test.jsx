import React from 'react';
import { render } from '@testing-library/react';
import { LoadingScreen } from './loading-screen';

describe('LoadingScreen Component', () => {

  // Matching Snapshots
  test('it should match the snapshot', () => {
    const { asFragment } = render(<LoadingScreen />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Unit tests

});