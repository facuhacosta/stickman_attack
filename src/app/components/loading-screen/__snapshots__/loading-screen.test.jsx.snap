// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingScreen Component it should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="undefined"
  />
</DocumentFragment>
`;
